areas/storage/src/AzureMcp.Storage/Commands/Blob/BlobDetailsCommand.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json.Serialization;
+using Azure.Storage.Blobs.Models;
+using AzureMcp.Core.Commands;
+using AzureMcp.Core.Services.Telemetry;
+using AzureMcp.Storage.Commands;
+using AzureMcp.Storage.Options;
+using AzureMcp.Storage.Options.Blob;
+using AzureMcp.Storage.Services;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.Storage.Commands.Blob;
+
+public sealed class BlobDetailsCommand(ILogger<BlobDetailsCommand> logger) : BaseBlobCommand<BlobDetailsOptions>()
+{
+    private const string CommandTitle = "Get Storage Blob Details";
+    private readonly ILogger<BlobDetailsCommand> _logger = logger;
+
+    public override string Name => "details";
+
+    public override string Description =>
+        $"""
+        Get blob properties, metadata, and general information. This tool retrieves blob configuration including metadata properties, 
+        approximate size, and last modification time information. Returns blob properties as JSON. Requires {StorageOptionDefinitions.AccountName}, 
+        {StorageOptionDefinitions.ContainerName}, and {StorageOptionDefinitions.BlobName}.
+        """;
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new() { Destructive = false, ReadOnly = true };
+
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        var options = BindOptions(parseResult);
+
+        try
+        {
+            if (!Validate(parseResult.CommandResult, context.Response).IsValid)
+            {
+                return context.Response;
+            }
+
+            context.Activity?.WithSubscriptionTag(options);
+
+            var storageService = context.GetService<IStorageService>();
+            var details = await storageService.GetBlobDetails(
+                options.Account!,
+                options.Container!,
+                options.Blob!,
+                options.Subscription!,
+                options.Tenant,
+                options.RetryPolicy
+            );
+
+            var result = new BlobDetailsCommandResult(new JsonBlobProperties(details));
+            context.Response.Results = ResponseResult.Create(result, StorageJsonContext.Default.BlobDetailsCommandResult);
+            return context.Response;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error getting blob details. Account: {Account}, Container: {Container}, Blob: {Blob}.", options.Account, options.Container, options.Blob);
+            HandleException(context, ex);
+            return context.Response;
+        }
+    }
+
+    internal record BlobDetailsCommandResult(JsonBlobProperties Details);
+
+    internal class JsonBlobProperties
```

Comment:
Curious why this is needed

Reply:
It probably isn't, I think this copied a pattern seen elsewhere in Storage that's likely unnecessary too.

Will have this as a follow-up to clean up all of Storage later.